app/recover_test.go

```diff
@@ -0,0 +1,26 @@
+package app
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRecover(t *testing.T) {
+	// Note: this test is not working as expected. It is not catching the panic.
+	// assert.NotPanics(t, func() {
+	// 	defer Recover()
+	// 	thisWillPanic()
+	// })
+
+	assert.NotPanics(t, func() {
+		go func() {
```

Comment:
This assert is not working as it should because of the asynchronously nature of go-routines. The `go` keyword will just create a new "thread" and leave it sitting there for the go runtime to start it and will continue processing the function.
Because it takes a few moments to spin up the go-routine and there is no more instructions in the current function, the current function ends, no panic happened yet and its assert passes. Then the program closes and no panic is ever registered. Your Recover function never executed and so it was never tested.
This can be made clear if you introduce a defect in the Recover function.
Please check this example: https://go.dev/play/p/S_qY9loDqCk
The recover is broken but it passes the first assert, because the test terminates before the go-routine is even started. But if we wait for the go-routine the panic will happen.